ibm/resource_ibm_is_placement_group.go

```diff
@@ -0,0 +1,467 @@
+				Computed:    true,
+				Description: "The lifecycle state of the placement group.",
+			},
+			"resource_type": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The resource type.",
+			},
+		},
+	}
+}
+
+func resourceIbmIsPlacementGroupValidator() *ResourceValidator {
+	validateSchema := make([]ValidateSchema, 1)
+	validateSchema = append(validateSchema,
+		ValidateSchema{
+			Identifier:                 "strategy",
+			ValidateFunctionIdentifier: ValidateAllowedStringValue,
+			Type:                       TypeString,
+			Required:                   true,
+			AllowedValues:              "host_spread, power_spread",
+		},
+		ValidateSchema{
+			Identifier:                 "name",
+			ValidateFunctionIdentifier: ValidateRegexpLen,
+			Type:                       TypeString,
+			Optional:                   true,
+			Regexp:                     `^([a-z]|[a-z][-a-z0-9]*[a-z0-9])$`,
+			MinValueLength:             1,
+			MaxValueLength:             63,
+		},
+		ValidateSchema{
+			Identifier:                 "tag",
+			ValidateFunctionIdentifier: ValidateRegexpLen,
+			Type:                       TypeString,
+			Optional:                   true,
+			Regexp:                     `^[A-Za-z0-9:_ .-]+$`,
+			MinValueLength:             1,
+			MaxValueLength:             128,
+		},
+		ValidateSchema{
+			Identifier:                 "accesstag",
+			ValidateFunctionIdentifier: ValidateRegexpLen,
+			Type:                       TypeString,
+			Optional:                   true,
+			Regexp:                     `^([ ]*[A-Za-z0-9:_.-]+[ ]*)+$`,
+			MinValueLength:             1,
+			MaxValueLength:             128,
+		},
+	)
+
+	resourceValidator := ResourceValidator{ResourceName: "ibm_is_placement_group", Schema: validateSchema}
+	return &resourceValidator
+}
+
+func resourceIbmIsPlacementGroupCreate(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vpcClient, err := meta.(ClientSession).VpcV1API()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	createPlacementGroupOptions := &vpcv1.CreatePlacementGroupOptions{}
+
+	createPlacementGroupOptions.SetStrategy(d.Get("strategy").(string))
+	createPlacementGroupOptions.SetName(d.Get("name").(string))
+
+	if resourceGroupIntf, ok := d.GetOk("resource_group"); ok && resourceGroupIntf.(string) != "" {
+		resourceGroup := resourceGroupIntf.(string)
+		resourceGroupIdentity := &vpcv1.ResourceGroupIdentity{
+			ID: &resourceGroup,
+		}
+		createPlacementGroupOptions.SetResourceGroup(resourceGroupIdentity)
+	}
+
+	placementGroup, response, err := vpcClient.CreatePlacementGroupWithContext(context, createPlacementGroupOptions)
+	if err != nil {
+		log.Printf("[DEBUG] CreatePlacementGroupWithContext failed %s\n%s", err, response)
+		return diag.FromErr(err)
+	}
+
+	d.SetId(*placementGroup.ID)
+
+	_, err = isWaitForPlacementGroupAvailable(vpcClient, d.Id(), d.Timeout(schema.TimeoutCreate), d)
+	if err != nil {
+		return diag.FromErr(err)
```

Comment:
Please add error strings in all wait errors
`return diag.FromErr(err)` ==> `return diag.FromErr(fmt.Errorf("[ERROR] Error waiting for placement group to be available %s", err)`